lib/driver.js

```diff
@@ -171,7 +171,9 @@ class EspressoDriver extends BaseDriver {
       return [sessionId, caps];
     } catch (e) {
       await this.deleteSession();
-      throw e;
+      e.message += '. If the cause might be espresso driver or server, ' +
+        'please read troubleshooting https://github.com/appium/appium-espresso-driver#contributing';
+      throw new errors.SessionNotCreatedError(e);
```

Comment:
I'd rather check if the error type is already `SessionNotCreatedError`. Also, the constructor argument should be `e.message` rather than `e`